packages/jest-core/src/SearchSource.ts

```diff
@@ -266,10 +266,19 @@ export default class SearchSource {
     } else if (globalConfig.runTestsByPath && paths && paths.length) {
       return this.findTestsByPaths(paths);
     } else if (globalConfig.findRelatedTests && paths && paths.length) {
-      return this.findRelatedTestsFromPattern(
+      const relatedTests = this.findRelatedTestsFromPattern(
         paths,
         globalConfig.collectCoverage,
       );
+      if (globalConfig.testPathPattern !== null) {
+        return {
+          tests: relatedTests.tests.filter(test =>
+            new RegExp(globalConfig.testPathPattern).test(test.path),
```

Comment:
I'd probably try to reuse the logic in `_filterTestPathsWithStats` on the related tests, hopefully this helps as a hint

Reply:
That is very helpful, I was wondering why was not seeing the stats output when 0 tests were matched. Making the update now!